lib/Sema/CSFix.h

```diff
@@ -632,32 +637,63 @@ class InsertExplicitCall final : public ConstraintFix {
                                     ConstraintLocator *locator);
 };
 
-class InsertPropertyWrapperUnwrap final : public ConstraintFix {
-  DeclName PropertyName;
+class UsePropertyWrapper final : public ConstraintFix {
+  VarDecl *Wrapped;
+  Optional<DeclName> MemberName;
+  bool UsingStorageWrapper;
   Type Base;
   Type Wrapper;
 
-  InsertPropertyWrapperUnwrap(ConstraintSystem &cs, DeclName propertyName,
-                              Type base, Type wrapper,
-                              ConstraintLocator *locator)
-      : ConstraintFix(cs, FixKind::InsertPropertyWrapperUnwrap, locator),
-        PropertyName(propertyName), Base(base), Wrapper(wrapper) {}
+  UsePropertyWrapper(ConstraintSystem &cs, VarDecl *wrapped,
+                     Optional<DeclName> memberName, bool usingStorageWrapper,
+                     Type base, Type wrapper, ConstraintLocator *locator)
+      : ConstraintFix(cs, FixKind::UsePropertyWrapper, locator),
+        Wrapped(wrapped), MemberName(memberName),
+        UsingStorageWrapper(usingStorageWrapper), Base(base), Wrapper(wrapper) {
+  }
 
 public:
   std::string getName() const override {
-    return "insert a $ to unwrap the property wrapper";
+    return "insert '$'s to use property wrapper type instead of wrapped type";
+  }
+
+  bool diagnose(Expr *root, bool asNote = false) const override;
+
+  static UsePropertyWrapper *create(ConstraintSystem &cs, VarDecl *wrapped,
+                                    Optional<DeclName> memberName,
+                                    bool usingStorageWrapper, Type base,
+                                    Type wrapper, ConstraintLocator *locator);
+};
+
+class UseWrappedValue final : public ConstraintFix {
+  VarDecl *PropertyWrapper;
+  Optional<DeclName> MemberName;
+  Type Base;
+  Type Wrapper;
+
+  UseWrappedValue(ConstraintSystem &cs, VarDecl *propertyWrapper,
+                  Optional<DeclName> memberName, Type base, Type wrapper,
+                  ConstraintLocator *locator)
+      : ConstraintFix(cs, FixKind::UseWrappedValue, locator),
+        PropertyWrapper(propertyWrapper), MemberName(memberName), Base(base),
+        Wrapper(wrapper) {}
+
+  bool usingStorageWrapper() const {
+    auto nameStr = PropertyWrapper->getName().str();
+    assert(!nameStr.empty() && nameStr[0] == '$');
+    return nameStr.size() > 1 && nameStr[1] != '$';
```

Comment:
I think you can  replace this with `startsWith("$$")` which could be  called on StringRef.